File: language-support/js/daml-ledger/src/model/completion_stream_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from '..';

/**
 * @interface CompletionStreamRequest
 * @memberof ledger
 */
export interface CompletionStreamRequest {
    /**
     * @member {string} applicationId
     * @memberof ledger.CompletionStreamRequest
     * @instance
     */
    applicationId: string
    /**
     * @member {ledger.LedgerOffset} offset
     * @memberof ledger.CompletionStreamRequest
     * @instance
     */
    offset: ledger.LedgerOffset
    /**
     * @member {Array<string>} parties
     * @memberof ledger.CompletionStreamRequest
     * @instance
     */
    parties: string[]
}

File: ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/messages/command/completion/CompletionStreamRequest.scala
// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.api.messages.command.completion

import com.daml.ledger.api.domain.{LedgerId, LedgerOffset}
import com.daml.lf.data.Ref

case class CompletionStreamRequest(
    ledgerId: Option[LedgerId],
    applicationId: Ref.ApplicationId,
    parties: Set[Ref.Party],
    offset: Option[LedgerOffset],
)
